bpf-loader-rs/bpf-loader-lib/src/export_event/tests/map_sampling_tests.rs

```diff
@@ -0,0 +1,359 @@
+ty<25> <ARRAY> n:16 idx-->[4] val-->[24]
+
+ */
+
+fn find_sample_map_mut(maps: &mut [MapMeta]) -> &mut MapMeta {
+    let mut sample_map = None;
+    for map in maps.iter_mut() {
+        if map.sample.is_some() {
+            sample_map = Some(map);
+        }
+    }
+    let sample_map = sample_map.unwrap();
+    sample_map
+}
+
+fn find_sample_map(maps: &[MapMeta]) -> &MapMeta {
+    let mut sample_map = None;
+    for map in maps.iter() {
+        if map.sample.is_some() {
+            sample_map = Some(map);
+        }
+    }
+    let sample_map = sample_map.unwrap();
+    sample_map
+}
+
+fn send_data(exporter: Arc<EventExporter>, key_buffer: &[u8], value_buffer: &[u8]) {
+    match &exporter.internal_impl {
+        ExporterInternalImplementation::KeyValueMapProcessor {
+            event_processor, ..
+        } => {
+            event_processor
+                .handle_event(key_buffer, value_buffer)
+                .unwrap();
+        }
+        _ => panic!("Unexpected internal implementation"),
+    };
+}
+
+fn create_exporter(
+    things: &LoadedThings,
+    handler: Arc<dyn EventHandler>,
+    export_format: ExportFormatType,
+) -> Arc<EventExporter> {
+    let meta = &things.package.meta;
+    let sample_map = find_sample_map(&meta.bpf_skel.maps[..]);
+
+    let exporter = EventExporterBuilder::new()
+        .set_export_event_handler(handler)
+        .set_export_format(export_format)
+        .build_for_map_sampling(
+            things.key_id,
+            things.value_id,
+            &sample_map.sample.as_ref().unwrap(),
+            &meta.export_types[..],
+            things.btf.clone(),
+        )
+        .unwrap();
+    exporter
+}
+
+#[test]
+fn test_export_format_json() {
+    let things = load_things();
+    let received_data = Rc::new(RefCell::new(String::default()));
+    struct MyEventHandler {
+        data: RRC<String>,
+    }
+    impl EventHandler for MyEventHandler {
+        fn handle_event(
+            &self,
+            _context: Option<Arc<dyn std::any::Any>>,
+            data: crate::export_event::ReceivedEventData,
+        ) {
+            match data {
+                crate::export_event::ReceivedEventData::JsonText(s) => {
+                    self.data.replace(s.to_string());
+                }
+                _ => panic!("Unexpected data type"),
+            }
+        }
+    }
+    let exporter = create_exporter(
+        &things,
+        Arc::new(MyEventHandler {
+            data: received_data.clone(),
+        }),
+        ExportFormatType::Json,
+    );
+    let (key_buffer, value_buffer) = create_key_value_buffer();
+    send_data(exporter, &key_buffer[..], &value_buffer[..]);
+    println!("{:?}", value_buffer);
+    let inner_data = received_data.borrow();
+    let decoded = serde_json::from_str::<HistsMap>(&inner_data).unwrap();
+    println!("{:?}", decoded);
+    decoded.verify_with_default_value();
+}
+
+const EXPECTED_OUTPUT_LOG2HISTS_LINES: [&str; 29] = [
```

Comment:
I feel that this test is not very necessary... (it looks odd, or rather funny)

Reply:
This test aims to test whether exporter of map-sampling can generate correct output for the pre-defined data. The hardcoded `EXPECTED_OUTPUT_LOG2HISTS_LINES` is the expected output of the data returned by `create_key_value_buffer`

Reply:
Then let's keep it, I have no problem with it.